src/Gui/CommandDoc.cpp

```diff
@@ -271,44 +272,221 @@ StdCmdExport::StdCmdExport()
     eType         = 0;
 }
 
+/**
+Create a default filename from a user-specified format string
+ 
+Format options are:
+%F - the basename of the .FCStd file (or the label, if it is not saved yet)
+%Lx - the label of the selected object(s), separated by character 'x'
+%Px - the label of the selected object(s) and their first parent, separated by character 'x'
+%U - the date and time, in UTC, ISO 8601
+%D - the date and time, in local timezone, ISO 8601
+Any other characters are treated literally, though if the filename is illegal
+it will be changed on saving.
+*/
+QString createDefaultExportBasename()

```

Comment:
Maybe mention that the formatstring  is stored in `exportFormatString`. 

My feeling is that this method is a bit too advanced for what it does. It is cool, but is there really a usecase for having date support etc? My doubt is if all the format options have been tested?

Reply:
We did discuss this a little in the forums, and using a custom format string is definitely a "power user" feature. I think the date is pretty handy for a sort of rudimentary "versioning" of exports during iterative design. I did test the format strings, but of course more testers to find edge cases would be very helpful for this PR.

I'll add some detail to that comment (and of course this will all get actually documented for end-users in the wiki if the PR is accepted).

Reply:
Fair enough.

Reply:
I find it an excellent strategy to have a simple version for "normal" users, and more powerful options that are hidden for that normal user, to not add complexity. These options should be explained with more detail somewhere (on the wiki basically), that is available for users who want to dig more.